### PR TITLE
Allow user to specify start time

### PR DESCRIPTION
**Pull request by rdooley**
* only use for recording spans
* yangxikun/opentelemetry-lua#86

### CODE REVIEW

File: lib/opentelemetry/trace/recording_span.lua [hunk 1]
```diff
@@ -23,13 +23,13 @@ local mt = {
 --                          config.attributes: span attributes
 -- @return              span
 ------------------------------------------------------------------
-function _M.new(tracer, parent_ctx, ctx, name, config)
+function _M.new(tracer, parent_ctx, ctx, name, config, start_time)
```
Comment (plantfansam) [inline lib/opentelemetry/trace/recording_span.lua:26]: Can we add arg to docstring above?

**Review by plantfansam — CHANGES_REQUESTED** [review]

File: lib/opentelemetry/trace/recording_span.lua (hunk 1 above)
Comment (rdooley) [reply lib/opentelemetry/trace/recording_span.lua:26]: oh of course, great catch

**Comment by rdooley** [comment]
how did the test pass because I forgot to alter the `tracer.start` method in the original commit? 🤔

**Comment by rdooley** [comment]
This i think needs some work, I will do additional lua unit testing as well

File: spec/trace/exporter/otlp_spec.lua [hunk 2]
```diff
@@ -36,6 +36,8 @@ describe("encode_spans", function()
         local resource = encoded.resource_spans[1]
         assert(#resource.instrumentation_library_spans == 1)
         assert(#resource.instrumentation_library_spans[1].spans == 10)
+        assert(resource.instrumentation_library_spans[1].spans[1].start_time_unix_nano == "123456788")
+        assert(resource.instrumentation_library_spans[1].spans[1].end_time_unix_nano == "123456789")
```
Comment (rdooley) [inline spec/trace/exporter/otlp_spec.lua:40]: Just a nice addition here to check that we are setting and exporting the times we expect on spans

File: lib/opentelemetry/trace/tracer.lua [hunk 3]
```diff
@@ -66,12 +66,13 @@ end
 -- @span_start_config   [optional]
 --                          span_start_config.kind: opentelemetry.trace.span_kind.*
 --                          span_start_config.attributes: a list of attribute
+-- @start_time          [optional] start time
```
Comment (plantfansam) [inline lib/opentelemetry/trace/tracer.lua:69]:
```suggestion
-- @start_time          [optional] start time: nanoseconds
```

**Review by plantfansam — APPROVED** [review]

**Pull request merged**